### PR TITLE
Link to user edit pages in importer messages

### LINKED ISSUE #1679: Link to user edit pages in importer messages
When the importer shows messages about users (such as "This new user has the same email as this existing user"), the respective users' edit pages should be linked in the message to enable managers to easily open the user profile for inspection.

### PR DESCRIPTION
**Pull request by jb3rndt**
Closes #1679

Preview:
grafik

First I had `user_edit_link` in the importer but then recognised that it is a pain forming the strings in the tests, so I moved it into tools. But im not sure if that is the right place for it

### CODE REVIEW

File: evap/staff/tools.py
Comment (janno42) [inline evap/staff/tools.py:376]:
```suggestion
        '<a href="{}" target=_blank><span class="fas fa-user-pen"> edit user</span></a>'.format(
```

(will work only after #1733 is merged)

**Review by janno42 — COMMENTED** [review]
✔️ Meets requirements
✔️ UI functionality checked

File: evap/staff/tests/test_views.py
Comment (He3lixxx) [inline evap/staff/tests/test_views.py:494]: Could you please add a unit test for user_edit_link? Currently, all tests just test that method against itself, so we wouldn't notice if it was broken.

File: evap/staff/tools.py
Comment (He3lixxx) [inline evap/staff/tools.py:379]:
```suggestion
    return format_html(
        '<a href="{}" target=_blank><span class="fas fa-user-pen"> {}</span></a>',
        reverse("staff:user_edit", kwargs={"user_id": user_id}),
        _("edit user"),
    )
```

**Review by He3lixxx — CHANGES_REQUESTED** [review]
Coverage reporting seems off -- will have to look at it when this is rebased properly.

File: evap/staff/tools.py
Comment (He3lixxx) [inline evap/staff/tools.py:380]: This `.format`-inside-`format_html` construction will "abuse" `format_html` as `mark_safe`. Both format arguments should be properly escaped.

```suggestion
    return format_html(
        '<a href="{}" target=_blank><span class="fas fa-user-pen"> {}</span></a>',
        reverse("staff:user_edit", kwargs={"user_id": user_id}),
        _("edit user"),
    )
```

**Review by He3lixxx — CHANGES_REQUESTED** [review]

File: evap/staff/importers.py
Comment (niklasmohrin) [inline evap/staff/importers.py:336]: I think this is equivalent to
```python
return format_html(
    "{}<br /> - {} ({}) [{}]<br /> - {} ({})",
    msg,
    ExcelImporter._create_user_string(user),
    _("existing"),
    user_edit_link(user.pk),
    ExcelImporter._create_user_string(user_data),
    _("new"),
)
```
Furthermore, you could remove the single argument `format_html` with `msg` above

Comment (He3lixxx) [reply evap/staff/importers.py:336]: Ahh yes good old `{}<br /> - {} ({}) [{}]<br /> - {} ({})`

**Review by He3lixxx — APPROVED** [review]

File: evap/staff/importers.py
Comment (jb3rndt) [reply evap/staff/importers.py:336]: Actually true ^^

File: evap/staff/tools.py
Comment (janno42) [inline evap/staff/tools.py:376]:
```suggestion
        '<a href="{}" target=_blank><span class="fas fa-user-pen"></span> {}</a>',
```
i'm sorry, my fault

**Review by janno42 — CHANGES_REQUESTED** [review]

File: evap/staff/importers.py
Comment (niklasmohrin) [inline evap/staff/importers.py:354]: Could you change this single-argument `format_html` to be a `mark_safe`?

**Review by niklasmohrin — COMMENTED** [review]
Code looks good, seems to work too, but I haven't manually tested in depth, I leave that to @janno42 :)

**Review by janno42 — APPROVED** [review]
✔️ Meets requirements
✔️ UI functionality checked

**Review by niklasmohrin — APPROVED** [review]

**Pull request merged**
